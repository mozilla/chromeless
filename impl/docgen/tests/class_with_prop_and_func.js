/**
 * A simple class with a property and a function
 */

/**
 * @class URL
 * A class which parses a url and exposes its various
 * components separately.
 */

/**
 * @property foo
 * the foo property of my url classs
 * @type string
 */

/**
 * @function bar
 * the foo property of my url classs
 * @returns nothing
 */

/** @endclass */
