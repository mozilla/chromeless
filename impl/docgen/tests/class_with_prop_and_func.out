{
  "classes": [
    {
      "desc": "A class which parses a url and exposes its various\ncomponents separately.", 
      "functions": [
        {
          "desc": "the foo property of my url classs", 
          "name": "bar", 
          "returns": {
            "desc": "nothing"
          }
        }
      ], 
      "name": "URL", 
      "properties": [
        {
          "desc": "the foo property of my url classs", 
          "name": "foo", 
          "type": "string"
        }
      ]
    }
  ], 
  "desc": "A simple class with a property and a function", 
  "filename": "tests/class_with_prop_and_func.js", 
  "module": "class_with_prop_and_func"
}
